Assets/Scripts/Calendar UI.cs

```diff
@@ -23,89 +25,136 @@ public class CalendarUI : MonoBehaviour
     [SerializeField] private TextMeshProUGUI yearText;
 
 
+
     private DateTime currentDate;
+    private DateTime defaultDate;
     private DateTime firstDayOfMonth;
+
     private int daysInMonth;
     private int numberOfBlanksBefore;
     private int numberOfBlanksAfter;
+
     private void Start()
     {
-
+        EventManager.instance.onForwardClick += HandleForwardClick;
+        EventManager.instance.onBackwardClick += HandleBackwardClick;
         currentDate = DateTime.Today;
+        defaultDate = currentDate;
+        monthText.text = defaultDate.ToString("MMMM");
+        yearText.text = defaultDate.ToString("yyyy");
+
 
-        monthText.text = currentDate.ToString("MMMM");
 
-        yearText.text = currentDate.ToString("yyyy");
-        firstDayOfMonth = new DateTime(currentDate.Year, currentDate.Month, 1);
-        daysInMonth = DateTime.DaysInMonth(currentDate.Year, currentDate.Month);
         numberOfBlanksBefore = GetBlanksBefore();
         numberOfBlanksAfter = GetBlanksAfter();
+        PopulateCalendarGrid();
         PopulateWeekDaysGrid();
+
+    }
+
+    private void OnDestroy()
+    {
+        EventManager.instance.onForwardClick -= HandleForwardClick;
+        EventManager.instance.onBackwardClick -= HandleBackwardClick;
+    }
+
+
+    private void HandleForwardClick()
+    {
+        calendarGridPopulator.ClearCalendarGrid();
+
+
+        defaultDate = defaultDate.AddMonths(1);
+        defaultDate = new DateTime(defaultDate.Year, defaultDate.Month, 1);
+
+        monthText.text = defaultDate.ToString("MMMM");
+        yearText.text = defaultDate.ToString("yyyy");
+        numberOfBlanksBefore = GetBlanksBefore();
+        numberOfBlanksAfter = GetBlanksAfter();
+
         PopulateCalendarGrid();
+
     }
+    private void HandleBackwardClick()
+    {
+        calendarGridPopulator.ClearCalendarGrid();
+        defaultDate = defaultDate.AddMonths(-1);
 
 
+        monthText.text = defaultDate.ToString("MMMM");
+        yearText.text = defaultDate.ToString("yyyy");
+        numberOfBlanksBefore = GetBlanksBefore();
+        numberOfBlanksAfter = GetBlanksAfter();
+        PopulateCalendarGrid();
+    }
 
     public void PopulateCalendarGrid()
     {
 
         int totalDays = numberOfBlanksBefore + daysInMonth + numberOfBlanksAfter;
-        List<GameObject> populationList = GridPopulator.instance.PopulateTheGrid(totalDays, calendarGrid, dayPrefab);
+        List<Cell> populationList = calendarGridPopulator.PopulateTheGrid(totalDays);
```

Comment:
Not a big deal but this function should maybe just be called something like Populate(). No only does this lend itself more to interface implementation, it implies there are other things to Populate inside of grid populator. 

gridPopulator.PopulateTheGrid();

vs 

gridPopulator.Populate();

Reply:
Resolved